lib/libc/math/lib_logf.c

```diff
@@ -81,17 +83,25 @@ float logf(float x)
 	iter = 0;
 	relax_factor = 1;
 
+	if (x < 0.0) {
+                return NAN;
+        }
+
+	if (x == 0.0) {
+                return -HUGE_VAL;
```

Comment:
I think its better to represent HUGE_VALF to indicate float value.

Reply:
It will be same.
But yes,I will do it.